src/common/transformations/src/transformations/common_optimizations/transpose_sinking_binary.cpp

```diff
@@ -0,0 +1,383 @@
+        auto transpose_node = ov::as_type_ptr<ov::opset9::Transpose>(input_node);
+        if (!transpose_node)
+            continue;
+        auto constant_node =
+            ov::as_type_ptr<ov::opset9::Constant>(transpose_node->input_value(1).get_node_shared_ptr());
+        if (!constant_node)
+            continue;
+        {
+            TrasposeInputsInfo input_info;
+            input_info.transpose = transpose_node;
+            input_info.transpose_const = constant_node;
+            input_info.input_idx = input_idx;
+            return input_info;
+        }
+    }
+
+    return TrasposeInputsInfo();
+}
+
+bool IfNodeHasTransposeInputs(const ov::Output<ov::Node>& output) {
+    TrasposeInputsInfo inputs_info = GetFirstTransposeInput(output.get_node_shared_ptr());
+    return !inputs_info.isEmpty();
+}
+
+// --------------------------------------------------------------------------------------
+
+ov::AxisVector ReverseTransposeOrder(const ov::AxisVector& axis_order) {
+    ov::AxisVector out(axis_order.size());
+    for (size_t i = 0; i < axis_order.size(); i++) {
+        out.at(axis_order[i]) = i;
+    }
+    return out;
+}
+
+}  // namespace
+
+void SwapOutputNames(ov::Output<ov::Node> output1, ov::Output<ov::Node> output2) {
+    const auto node2_output_names = output2.get_names();
+    output2.set_names(output1.get_names());
+    output1.set_names(node2_output_names);
+}
+
+void SwapFriendlyNames(NodePtr node1, NodePtr node2) {
+    const std::string node2_name = node2->get_friendly_name();
+    node2->set_friendly_name(node1->get_friendly_name());
+    node1->set_friendly_name(node2_name);
+}
+
+void SwapNames(NodePtr node1, NodePtr node2) {
+    SwapFriendlyNames(node1, node2);
+    SwapOutputNames(node1->output(0), node2->output(0));
+}
+
+// --------------------------------------------------------------------------------------
+
+namespace {
+
+// get new axis for Concat/Split according to transpose order
+template <typename T>
+T TransposeAxis(T axis, const ov::AxisVector& transpose_order) {
+    return transpose_order[axis];
+}
+
+}  // namespace
+
+// --------------------------------------------------------------------------------------
+
+struct OutputTranspose {
+    OutputTranspose() : transpose(nullptr), transpose_const(nullptr) {}
+    ov::opset9::Transpose* transpose;
+    ov::opset9::Constant* transpose_const;
+};
+
+OutputTranspose GetOutputTransposes(NodePtr node) {
+    for (size_t output_idx = 0; output_idx < node->get_output_size(); ++output_idx) {
+        for (auto& input : node->get_output_target_inputs(output_idx)) {
+            auto transpose_node = dynamic_cast<ov::opset9::Transpose*>(input.get_node());
+            if (!transpose_node)
+                continue;
+            auto constant_node = dynamic_cast<ov::opset9::Constant*>(transpose_node->input_value(1).get_node());
+            if (!constant_node)
+                continue;
+            {
```

Comment:
not sure that we need this body

Reply:
Why? We need to find Transpose and get constant value for sinking.